functions/src/create_friend_invitation_link.ts

```diff
@@ -0,0 +1,108 @@
+const auth = admin.auth();
+
+const corsHandler = cors({ origin: true });
+
+/* This endpoint is called when a user wants to send an invitation*/
+export const createFriendInvitationLink = functions.https.onRequest(
+  async (request, response) => {
+    corsHandler(request, response, async () => {
+      try {
+        try {
+          // This request should be made with content type is application/x-www-form-urlencoded.
+          // This is done to prevent a pre-flight CORS request made to the firebase function
+          // Refer: https://github.com/gt-scheduler/website/pull/187#issuecomment-1496439246
+          request.body = JSON.parse(request.body);
+        } catch {
+          response.status(401).json(apiError("Bad request"));
+        }
+        const { IDToken, term, versions, redirectURL, validFor } =
+          request.body as CreateInviteRequestData;
+
+        if (!IDToken) {
+          return response.status(401).json(apiError("IDToken not provided"));
+        }
+        if (!term || !versions || !redirectURL || !validFor) {
+          return response
+            .status(400)
+            .json(apiError("Invalid arguments provided"));
+        }
+
+        // Authenticate token id
+        let decodedToken: admin.auth.DecodedIdToken;
+        try {
+          decodedToken = await auth.verifyIdToken(IDToken);
+        } catch {
+          return response.status(401).json(apiError("User not found"));
+        }
+
+        const senderEmail = decodedToken.email;
+
+        if (!senderEmail) {
+          return response
+            .status(400)
+            .json(apiError("Cannot share schedule link without an email"));
+        }
+
+        // Get Sender UID from the decoded token
+        const senderId = decodedToken.uid;
+
+        // Get Sender record from the schedules collection - it has to be version 3 because an invite was sent from it
+        const senderRes = await schedulesCollection.doc(senderId).get();
+        const senderData: Version3ScheduleData | undefined =
+          senderRes.data() as Version3ScheduleData | undefined;
+
+        if (
+          !senderData ||
+          !senderData?.terms ||
+          !senderData.terms[term]?.versions ||
+          versions.filter((v) => !senderData.terms[term].versions[v]).length > 0
+        ) {
+          return response
+            .status(400)
+            .json(apiError("Cannot make link for invalid schedule version"));
+        }
+
+        // find and delete existing links for the same sender, term, and version
+        const existingInvites = await invitesCollection
+          .where("sender", "==", senderId)
+          .where("term", "==", term)
+          .where("versions", "==", versions)
+          .where("link", "==", true)
+          .where("validFor", "==", validFor)
+          .get();
+        const batch = firestore.batch();
+        existingInvites.forEach((doc) => {
+          batch.delete(doc.ref);
+        });
+        await batch.commit();
+
```

Comment:
apologies for the confusion, we still need to create the invite record. It's just the `schedulesCollection.doc(senderId).set(senderData);` that is not required since we haven't made any changes to the sender's data. (Only changes in the invites collection are needed)